js/catalog-search.js

```diff
@@ -4,23 +4,25 @@ bcpl.catalogSearch = (($, queryStringer, waitForExistence, constants) => {
 	const catalogSearchSelector = '#catalog-search, .catalog-search';
 	const resultsInfoContainerSelector = '.gsc-above-wrapper-area-container';
 	const searchCatalogButton = '<td><button id="catalog-search" class="btn btn-primary pull-right">Search the Catalog</button></td>';
-
+	const showResults = "#resInfo-0:contains('About')";
 	const getCatalogUrl = (searchTerm) => `${constants.baseCatalogUrl}${constants.search.urls.catalog}${searchTerm}`;
 
-	const onCatalogSearchClick = (clickEvent) => {
-		clickEvent.preventDefault();
-
+	const getSearchTerm = () =>{
 		const queryParams = queryStringer.getAsDictionary();
-		const searchTerm = queryParams.search;
+		return queryParams.term;
+	};
 
-		window.location = getCatalogUrl(searchTerm);
+	const onCatalogSearchClick = (clickEvent) => {
+		clickEvent.preventDefault();
+		window.location = getCatalogUrl(getSearchTerm());
 	};
 
 	const init = () => {
-		waitForExistence(resultsInfoContainerSelector, () => {
+		waitForExistence(showResults, () => {
```

Comment:
I'm not sure this is the right approach.

I would leave the wait for existence the same and try something like this.

```js
$(resultsInfoContainerSelector)
  .find("td")
  .first()
  .prepend(`<span class="show-more-results-msg">Showing results for <em>${getSearchTerm()}</em></span>`)
  .after(searchCatalogButton);
```

This will remove the need for a new selector. I added added some markup because I have a feeling, well want to style this a bit.

Does this make sense?

Reply:
Makes perfect sense. Code updated to reflect suggested code changes.